devito/operator.py

```diff
@@ -96,6 +96,17 @@ def expr_cse(expr):
                     s_dict[arg] = Indexed(to_revert[value.base.label], *value.indices)
         to_revert[temp] = value.xreplace(s_dict)
 
+    # Make sure that no temporary that has to be reverted depends
+    # on another temporary of the list
+    changing = True
+    while changing:
+        changing = False
+        for temp, value in to_revert.items():
+            n_value = value.xreplace(to_revert)
```

Comment:
I replace this line by :
```
            try:
                n_value = value.xreplace(to_revert)
            except:
                to_keep.append((temp, value))
                to_revert.pop(temp, None)
```
And everything runs fine I am just not sure this is how it should be done and how risky it is